cfgen/cfgen

```diff
@@ -819,6 +820,32 @@ def fabricate_disks(hostname: str, paths: List[str]) -> List[Disk]:
     return [Disk(path=x, size=0, blksize=4096) for x in paths]
 
 
+def set_log_disks_size(log_disks: List[Disk], host: str) -> None:
+    """
+    Update the log disk size provided by the user/lsblk based on the
+    motr constraints.
+    """
+    def powOf2(num: int) -> int:
+        power = int(lg(num, 2))
+        return int(pow(2, power))
+
+    KB = int(1024)
+    MB = int(KB ** 2)
+    GB = int(KB ** 3)
+    # [TODO] Hardcoded constraints based on motr requirements.
+    upper_bound = 4*GB
+    lower_bound = 128*MB
+
+    # Rounding of log device size to the nearest(lower) power of 2.
+    for i, disk in enumerate(log_disks):
+        size = powOf2(disk.size)
+        _assert(size >= lower_bound, f"Node {host}: Log device size "
```

Comment:
I think instead of asserting we should bail out cleanly by logging an error message.

Reply:
Do you mean by raising an exception or log and exit(0)?